NuRadioReco/examples/PhasedArray/SNR_curves/T02RunSNR_RNOG.py

```diff
@@ -0,0 +1,280 @@
+                                                     trigger_adc=False, # Don't have a seperate ADC for the trigger
+                                                     adc_output='voltage', # output in volts
+                                                     trigger_filter=None,
+                                                     upsampling_factor=upsampling_factor,
+                                                     window=int(32 * new_sampling_rate * upsampling_factor / 2),
+                                                     step = int(16 * new_sampling_rate * upsampling_factor / 2))
+            else:
+                original_traces_ = self._station.get_channel(4).get_trace()
+
+                squared_mean, num_frames = triggerSimulator.powerSum(coh_sum=original_traces_,
+                                                                     window=int(32 * new_sampling_rate / 2),
+                                                                     step=int(16 * new_sampling_rate / 2),
+                                                                     adc_output=f'voltage')
+
+                squared_mean_threshold = np.power(threshold, 2.0)
+
+                has_triggered = (True in (squared_mean > squared_mean_threshold))
+
+            if(has_triggered):
+                print('Trigger on noise... ')
+
+        filtered_noise_traces = {}        
+        for channel in self._station.iter_channels():
+            filtered_noise_traces[channel.get_id()] = channel.get_trace()            
+
+        for factor, iSNR in zip(mult_factors, range(len(mult_factors))):
+
+            for channel in self._station.iter_channels():
+                trace = copy.deepcopy(filtered_signal_traces[channel.get_id()][:]) * factor
+                noise = filtered_noise_traces[channel.get_id()]
+                channel.set_trace(trace + noise, sampling_rate=new_sampling_rate)
+
+            if(phase):
+                has_triggered = triggerSimulator.run(self._evt, self._station, self._det,
+                                                     Vrms = 1.0,
+                                                     threshold = threshold,
+                                                     triggered_channels=channels,
+                                                     phasing_angles=phasing_angles,
+                                                     ref_index = 1.75,
+                                                     trigger_name='primary_phasing',
+                                                     trigger_adc=False,
+                                                     adc_output='voltage',
+                                                     trigger_filter=None,
+                                                     upsampling_factor=upsampling_factor,
+                                                     window=int(32 * new_sampling_rate * upsampling_factor / 2),
```

Comment:
why do you divide by 2 here?

Reply:
That is a hold-over from a previous integration window I was studying (32 ns vs 16 ns), I'll make sure to remove it for clarity.